### ISSUE TITLE
Ensure newline at end of file

### ISSUE THREAD

**Issue by takluyver** [spyder-ide/spyder#2759]
Many editors have an option to ensure that the file ends with a newline. This tends to make nicer VCS diffs if you add to the end of the file, and some linting tools will also complain if the file doesn't end in a newline. It would be good to have this option in Spyder too.

Apologies if I've missed either the option or an existing issue for this - I have tried to look for both.

**Comment by ccordoba12** [comment]
Thanks for the suggestion Thomas. I've marked for 3.1 because we still have a lot to do for 3.0 :-)

**Comment by goanpeca** [comment]
Is the idea here to have an option that will check if the file ends in a new line and if not, add it on save?

**Comment by ccordoba12** [comment]
Yes. that's the idea.

**Comment by maurerle** [comment]
This exists and can be set in the options:

- Editor->Source Code->Insert a newline at the end if one does not exist when saving a file

This issue can be closed.

**Comment by ccordoba12** [comment]
Right, that's been available since Spyder 4.2.0, I think. Closing then.
